postreise/plot/plot_bar_generation_stack.py

```diff
@@ -0,0 +1,234 @@
+    :param dict t2hc: user specified color of curtailable resource hatches to overwrite
+        type2hatchcolor default dict. key: resource type, valid keys are
+        *'curtailment'*, *'solar_curtailment'*, *'wind_curtailment'*,
+        *'wind_offshore_curtailment'*, value: color code.
+    :param dict titles: user specified figure titles, key: area, value: new figure
+        title in string, use area as title if None.
+    :param bool plot_show: save the generated figure or not, defaults to True.
+    :param bool save: save the generated figure or not, defaults to False.
+    :param dict filenames: user specified filenames, key: area, value: new filename
+        in string, use area as filename if None.
+    :param str filepath: if save is True, user specified filepath, use current
+        directory if None.
+    :return: (*list*) -- matplotlib.axes.Axes object of each plot in a list.
+    :raises TypeError:
+        if resources is not a list and/or
+        if titles is provided but not in a dictionary format and/or
+        if filenames is provided but not in a dictionary format.
+    :raises ValueError:
+        if length of area_types and areas is different and/or
+        if length of scenario_names and scenario_ids is different.
+    """
+    if isinstance(areas, str):
+        areas = [areas]
+    if not scenario_ids:
+        scenario_ids = []
+    if isinstance(scenario_ids, (int, str)):
+        scenario_ids = [scenario_ids]
+    if isinstance(resources, str):
+        resources = [resources]
+    if not isinstance(resources, list):
+        raise TypeError("ERROR: resources should be a list")
+    if isinstance(area_types, str):
+        area_types = [area_types]
+    if not area_types:
+        area_types = [None] * len(areas)
+    if len(areas) != len(area_types):
+        raise ValueError(
+            "ERROR: if area_types are provided, it should have the same number of entries with areas"
+        )
+    if isinstance(scenario_names, str):
+        scenario_names = [scenario_names]
+    if scenario_names and len(scenario_names) != len(scenario_ids):
+        raise ValueError(
+            "ERROR: if scenario names are provided, number of scenario names must match number of scenario ids"
+        )
+    if titles is not None and not isinstance(titles, dict):
+        raise TypeError("ERROR: titles should be a dictionary if provided")
+    if filenames is not None and not isinstance(filenames, dict):
+        raise TypeError("ERROR: filenames should be a dictionary if provided")
+    s_list = []
+    for sid in scenario_ids:
+        s_list.append(Scenario(sid))
+    mi = ModelImmutables(s_list[0].info["grid_model"])
+    type2color = mi.plants["type2color"]
+    type2label = mi.plants["type2label"]
+    type2hatchcolor = mi.plants["type2hatchcolor"]
+    if t2c:
+        type2color.update(t2c)
+    if t2l:
+        type2label.update(t2l)
+    if t2hc:
+        type2hatchcolor.update(t2hc)
```

Comment:
Are we going to compare scenarios using different grid model? If yes, we could move the above lines (starting where we create an instance of `ModelImmutables`) in the `for` loop on the scenario to get the each scenario grid model pre-defined `type2color`, etc.

Reply:
Good point. We can definitely compare scenarios created by different models for this figure. Will do so.

Reply:
@rouille Actually, this might cause implicit conflicts as well. For example, we have two scenarios created by `usa_tamu` and `hifld` model respectively. They have different default colors for 'wind' resource. If we put two such scenarios in a list to show the generation stack bars side by side, the colors are going to be overwritten by the model of the last scenario in the list.

Reply:
Given `mi` here is just for loading default color dicts, I suggest to keep it in its current form, i.e. always pick the default colors for the model of the first scenario in the list. We can still have scenarios with different models side by side using the default color scheme defined by the first scenario (the user can overwrite them all by these keyword arguments) as long as they have same areas which won't break `area_to_loadzone` as you mentioned below, otherwise, an "invalid area" error will be raised.